.github/workflows/check-kernel-config.yml

```diff
@@ -0,0 +1,74 @@
+# Licensed under the MIT License.
+
+# This action checks that required kernel configs have not been removed or
+# modified to an undesirable value. It also checks that new configs added to
+# the kernel config files are in the list of required configs.
+name: Kernel Required Configs Check
+
+on:
+  push:
+    branches: [main, dev, 1.0*, 2.0*]
+  pull_request:
+    branches: [main, dev, 1.0*, 2.0*]
+
+jobs:
+  check:
+    name: Kernel configs check
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout the branch of our repo that triggered this action
+      - name: Workflow trigger checkout
+        uses: actions/checkout@v2
+        
+      - name: Get base commit for PRs
+        if: ${{ github.event_name == 'pull_request' }}
+        run: |
+          git fetch origin ${{ github.base_ref }}
+          echo "base_sha=$(git rev-parse origin/${{ github.base_ref }})" >> $GITHUB_ENV
+          echo "Merging ${{ github.sha }} into ${{ github.base_ref }}"
+
+      - name: Get base commit for Pushes
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          git fetch origin ${{ github.event.before }}
+          echo "base_sha=${{ github.event.before }}" >> $GITHUB_ENV
+          echo "Merging ${{ github.sha }} into ${{ github.event.before }}"
+
+
+      # For consistency, we use the same major/minor version of Python that CBL-Mariner ships
+      - name: Setup Python 3.7
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+
+      - name: Get Python dependencies
+        run: python3 -m pip install -r toolkit/scripts/requirements.txt
+
+      # Check kernel configs
+      - name: Get the changed files and configs
+        run: |
+          echo "Files changed: '$(git diff-tree --no-commit-id --name-only -r ${{ env.base_sha }} ${{ github.sha }})'"
+          changed_configs=$(git diff-tree --diff-filter=d  --no-commit-id --name-only -r ${{ env.base_sha }} ${{ github.sha }} | { grep "SPECS/kernel/config.*$" || test $? = 1; })
+          echo "Files to validate: '${changed_configs}'"
+          echo "updated-configs=$(echo ${changed_configs})" >> $GITHUB_ENV
+
+          # Check for new configs in the PR
+          # and check if they are in required configs
+          if [[ -n $changed_configs ]]; then
+            config_changes=$(git diff-tree -p -r ${{ env.base_sha }} ${{ github.sha }} -- SPECS/kernel/config)
+            echo "AMD config changes:"
+            python3 .github/workflows/check_new_kernel_configs.py --required_configs=toolkit/scripts/mariner-required-configs.json --config_str="${changed_configs}"
+            
+            echo "ARM config changes:"
+            config_aarch64_changes=$(git diff-tree -p -r ${{ env.base_sha }} ${{ github.sha }} -- SPECS/kernel/config_aarch64)
+            python3 .github/workflows/check_new_kernel_configs.py --required_configs=toolkit/scripts/mariner-required-configs.json --config_str="${config_aarch64_changes}"
```

Comment:
>check_new_kernel_configs [](http://example.com/codeflow?start=38&length=24)

Do we really want to catch every new config entry? With newer kernels there's going to be newer configs and for some of them we're going to take the default value because we just don't have any specific requirements. At least that's my guess. If our policy is to track every single config entry, then I think this is fine, but otherwise, we're going to bloat that JSON file and potentially cause a lot of headache for people without a good reason.

Related question: what about config entries, which were there in the past and now a config update is robust enough, for git to think their lines got removed and added elsewhere? In such cases we're going to flag an pre-existing config as a new one.

Reply:
This was an ask from Chris to ensure the required configs stay updated.

Reply:
As long as Chris, you, and everyone else involved in managing the kernel accept this additional workload, then who am I to stop you?:)